checker/src/main/java/org/checkerframework/checker/index/samelen/SameLenTransfer.java

```diff
@@ -249,4 +256,62 @@ AnnotationMirror getAnno(Node n) {
 
         return new ConditionalTransferResult<>(result.getResultValue(), thenStore, elseStore);
     }
+
+    /** Overridden to ensure that SameLen annotations on method parameters are symmetric. */
+    @Override
+    protected void addInformationFromPreconditions(
+            CFStore info,
+            AnnotatedTypeFactory factory,
+            UnderlyingAST.CFGMethod method,
+            MethodTree methodTree,
+            ExecutableElement methodElement) {
+        super.addInformationFromPreconditions(info, factory, method, methodTree, methodElement);
+
+        List<? extends VariableTree> paramTrees = methodTree.getParameters();
+        List<String> paramNames = new ArrayList<>();
+        List<AnnotatedTypeMirror> params = new ArrayList<>();
+
+        for (VariableTree tree : paramTrees) {
+            paramNames.add(tree.getName().toString());
+            params.add(aTypeFactory.getAnnotatedType(tree));
+        }
+
+        for (int index = 0; index < params.size(); index++) {
+
+            // if the parameter has a samelen annotation, then look
+            // for other parameters in that annotation and propagate
+            // default the other annotation so that it is symmetric
+            AnnotatedTypeMirror atm = params.get(index);
+            AnnotationMirror anm = atm.getAnnotation(SameLen.class);
+            if (anm == null) {
+                continue;
+            }
+
+            List<String> values = IndexUtil.getValueOfAnnotationWithStringArgument(anm);
+            for (String value : values) {
+                int otherParamIndex = paramNames.indexOf(value);
+                if (otherParamIndex == -1) {
+                    continue;
+                }
+
+                // the SameLen value is in the list of params, so modify the type of
+                // that param in the store
+                AnnotationMirror newSameLen =
+                        aTypeFactory.createSameLen(
+                                Collections.singletonList(paramNames.get(index)));
+                Receiver otherParamRec = null;
+                try {
+                    otherParamRec =
+                            aTypeFactory.getReceiverFromJavaExpressionString(
```

Comment:
You have the tree, so you should be able to get the receiver from that.  Try 
https://checkerframework.org/api/org/checkerframework/framework/util/FlowExpressionParseUtil.html#internalReprOfVariable-org.checkerframework.framework.type.AnnotatedTypeFactory-com.sun.source.tree.VariableTree-